src/extension.ts

```diff
@@ -43,8 +43,11 @@ export async function activate(context: vscode.ExtensionContext) {
 		// Register tab change event
 		vscode.window.tabGroups.onDidChangeTabs(changeEvent => {
 			try {
+				const tab_data = vscode.window.tabGroups.all;
+				// Discard event if file is changed 
+				if (changeEvent.changed.length > 0) return;
 				const handler = new eventHandler(repoPath);
-				handler.handleTabChangeEvent(changeEvent);
+				handler.handleTabChangeEvent(tab_data);
```

Comment:
Maybe we should get Tabs data inside `handleTabChangeEvent` and should call `handleTabChangeEvent` only when in 2 cases
1. When a repo is opened
2. When TabChangeEvent is fired with closed/opened properties

Reply:
Regarding 1. - So bring our logic of discarding event if no repo is open in `extension.ts`, code will look something like: 

![image](https://github.com/codesyncapp/codesync-vs/assets/15974895/b98996ca-b435-463b-8ac7-dad6971a5866)

Reply:
How can we tackle the case: "user opens IDE (with pre-existing tabs) and starts coding" then? 

Bec I think according to this implementation, their current state of tabs won't be sent to our server.

Reply:
1. We can discard event for no repoPath even before changeEvent check
2. "How can we tackle the case: "user opens IDE (with pre-existing tabs) and starts coding" then?"
activate() is called when IDE is opened, or a repo is opened, we can call our eventHandler directly in `activate()` to capture initial tabs

Reply:
Okay got it, so for first time our handler will be run from `activate()`